settings.php

```diff
@@ -104,75 +104,83 @@
         }
     }
 
+    $warningtext = '';
     $signingsupport = false;
     if (!empty($config->filesystem)) {
         $signingsupport = (new $config->filesystem())->supports_presigned_urls();
     }
-    $warning = !method_exists('file_system', 'supports_xsendfile');
-    $warningtext = $warning ? $OUTPUT->notification(get_string('settings:presignedurl:coresupport', 'tool_objectfs')) : '';
+    if (!method_exists('file_system', 'supports_xsendfile')) {
+        $warningtext .= $OUTPUT->notification(get_string('settings:presignedurl:coresupport', 'tool_objectfs'));
+    }
     $warningtext .= \tool_objectfs\local\manager::cloudfront_pem_exists();
+    $classexists = class_exists('admin_setting_filetypes');
+    if (!class_exists('admin_setting_filetypes')) {
```

Comment:
These two lines are the same, did you meant o use $classexists?

Reply:
Ah, yes, initially it was `if (!$classexists) {`

Reply:
Fixed